packages/transport/src/modules/relationshipTemplates/RelationshipTemplateController.ts

```diff
@@ -227,12 +271,22 @@ export class RelationshipTemplateController extends TransportController {
         return template;
     }
 
-    public async loadPeerRelationshipTemplateByTruncated(truncated: string): Promise<RelationshipTemplate> {
+    public async loadPeerRelationshipTemplateByTruncated(truncated: string, password?: string, pin?: string): Promise<RelationshipTemplate> {
         const reference = RelationshipTemplateReference.fromTruncated(truncated);
-        return await this.loadPeerRelationshipTemplate(reference.id, reference.key, reference.forIdentityTruncated);
+        if (reference.passwordType?.startsWith("pw") && !password) throw TransportCoreErrors.general.noPasswordProvided();
+        if (reference.passwordType?.startsWith("pin") && !pin) throw TransportCoreErrors.general.noPINProvided();
+
+        return await this.loadPeerRelationshipTemplate(reference.id, reference.key, reference.forIdentityTruncated, password, pin, reference.salt);
     }
 
-    public async loadPeerRelationshipTemplate(id: CoreId, secretKey: CryptoSecretKey, forIdentityTruncated?: string): Promise<RelationshipTemplate> {
+    public async loadPeerRelationshipTemplate(
+        id: CoreId,
+        secretKey: CryptoSecretKey,
+        forIdentityTruncated?: string,
+        password?: string,
+        pin?: string,
+        salt?: CoreBuffer
+    ): Promise<RelationshipTemplate> {
```

Comment:
I think this function should be private as we heavily advise against loading a template by id and key.

Reply:
It's used when loading a template with a token

Reply:
Then you should add a function that allows to pass a TokenContentRelationshipTemplate instead.